server/routerlicious/packages/lambdas/src/deli/checkpointManager.ts

```diff
@@ -44,35 +45,72 @@ export interface ICheckpointParams {
 }
 
 export function createDeliCheckpointManagerFromCollection(
-	tenantId: string,
-	documentId: string,
-	collection: ICollection<IDocument>,
-): IDeliCheckpointManager {
-	const checkpointManager = {
-		writeCheckpoint: async (checkpoint: IDeliState) => {
-			return collection.update(
-				{
-					documentId,
-					tenantId,
-				},
-				{
-					deli: JSON.stringify(checkpoint),
-				},
-				null,
-			);
-		},
-		deleteCheckpoint: async () => {
-			return collection.update(
-				{
-					documentId,
-					tenantId,
-				},
-				{
-					deli: "",
-				},
-				null,
-			);
-		},
-	};
-	return checkpointManager;
+    tenantId: string,
+    documentId: string,
+    collection: ICollection<IDocument>,
+    localCollection: ICollection<ICheckpoint>): IDeliCheckpointManager {
+    const checkpointManager = {
+        writeCheckpoint: async (checkpoint: IDeliState, isLocal: boolean) => {
+            return isLocal ? localCollection.upsert({
```

Comment:
quick q: why does this have to be upsert for local and update for global?

Reply:
We clear the local collection when the last client disconnects, so if clients connect to the same document, there may be no checkpoint in the collection to update, so we will insert the new entry. In global we do not clear the documents collection so we can just update the existing entry